source/agora/test/Simple.d

```diff
@@ -44,3 +44,28 @@ unittest
     nodes.all!(node => node.getBlockHeight() == 1)
         .retryFor(2.seconds);
 }
+
+/// A test that stops and restarts a node
+unittest
+{
+    auto network = makeTestNetwork(TestConf.init);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    auto txes = makeChainedTransactions(getGenesisKeyPair(), null, 1);
+    txes.each!(tx => node_1.putTransaction(tx));
+
+    nodes.all!(node => node.getBlockHeight() == 1)
+        .retryFor(2.seconds);
+
+    // Now shut down & restart one node
+    auto restartMe = nodes[$-1];
+    network.restart(restartMe);
+    network.waitForDiscovery();
+    retryFor(restartMe.getBlockHeight() == 1, 5.seconds);
```

Comment:
How about changing the line as follows or adding this?
```
nodes.all!(node => node.getBlockHeight() == 1)
        .retryFor(2.seconds);
```
It depends on you.

Reply:
Makes sense, done